pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserDeleteService.java

```diff
@@ -0,0 +1,381 @@
+     * who deletes this user, not having status ENROLLED.
+     */
+    public void simpleDelete(Handle handle, User user, String whoDeleted, String comment) throws IOException {
+        delete(handle, user, whoDeleted, comment, false);
+    }
+
+    /**
+     * User full delete (including auth0 data). Cannot be deleted if have governed users (such users should
+     * be deleted first).
+     */
+    public void fullDelete(Handle handle, User user, String whoDeleted, String comment) throws IOException {
+        delete(handle, user, whoDeleted, comment, true);
+    }
+
+    private void delete(Handle handle, User user, String whoDeleted, String comment, boolean fullDelete) throws IOException {
+        LOG.info("User {} deletion is STARTED. guid:{}", fullDelete ? "FULL" : "SIMPLE", user.getGuid());
+        if (fullDelete) {
+            checkBeforeDelete(handle, user);
+        }
+        LogInfo logInfo = deleteUserSteps(handle, user, fullDelete);
+        LOG.warn("User {} deletion is completed SUCCESSFULLY. "
+                + "guid:{}, hruid:{}, e-mail:{}, name:{} {}, dob:{}\n"
+                + "Who deleted: {}\n"
+                + "Comment: {}\n"
+                + "Deleted data:\n{}",
+                fullDelete ? "FULL" : "SIMPLE",
+                user.getGuid(),
+                user.getHruid(),
+                user.getEmail(),
+                user.getProfile() != null ? objToStr(user.getProfile().getFirstName()) : "",
+                user.getProfile() != null ? objToStr(user.getProfile().getLastName()) : "",
+                user.getProfile() != null ? objToStr(user.getProfile().getBirthDate()) : "",
+                objToStr(whoDeleted),
+                objToStr(comment),
+                logInfo.getInfo());
+    }
+
+    private void checkBeforeDelete(Handle handle, User user) {
+        if (hasGovernedUsers(handle, user.getGuid())) {
+            throw new DDPException(format(ERROR_PREFIX_USER_DELETE + "the user has governed users", user.getGuid()));
+        }
+
+        // check if user refers to revisions
+        long[] revisionIds = handle.attach(JdbiRevision.class).findByUserId(user.getId());
+        if (revisionIds.length > 0) {
+            throw new DDPException(format(ERROR_PREFIX_USER_DELETE + "the user has references to a revision history", user.getGuid()));
+        }
+    }
+
+    private LogInfo deleteUserSteps(Handle handle, User user, boolean fullDelete) throws IOException {
+        LogInfo logInfo = new LogInfo();
+        UserCollectedData userCollectedData = new UserCollectedData();
+
+        if (fullDelete) {
+            deleteKitRequests(handle, user, logInfo);
+            deleteUserStudyLegacyData(handle, user, logInfo);
+            deleteUserAnnouncement(handle, user, logInfo);
+            deleteActivityInstanceStatusByOperator(handle, user, logInfo);
+            deleteAnswersByOperator(handle, user, logInfo);
```

Comment:
Hmm. Made comment about not needing this deleteAnswersByOperator elsewhere. github can be confusing....